src/openrct2/management/Finance.cpp

```diff
@@ -347,10 +342,9 @@ void game_command_set_current_loan(sint32 * eax, sint32 * ebx, sint32 * ecx, sin
 
     if (*ebx & GAME_COMMAND_FLAG_APPLY)
     {
-        money -= loanDifference;
-        gBankLoan      = newLoan;
-        gInitialCash   = money;
-        gCashEncrypted = ENCRYPT_MONEY(money);
+        gCash       -= loanDifference;
+        gBankLoan    = newLoan;
+        gInitialCash = money;
```

Comment:
Should be ` gInitialCash = gCash;`, and L335 should be `if (loanDifference > gCash)`. You can also remove the `money` variable on L317.